runelite-client/src/main/java/net/runelite/client/plugins/blastfurnace/BlastFurnacePlugin.java

```diff
@@ -135,4 +158,27 @@ public void onGameStateChanged(GameStateChanged event)
 			barDispenser = null;
 		}
 	}
+
+	@Subscribe
+	public void onGameTick(GameTick tick)
+	{
+		Widget npcDialog = client.getWidget(WidgetInfo.DIALOG_NPC_TEXT);
+
+		if (foremanCounter != null && foremanCounter > 0)
+		{
+			foremanCounter--;
+		}
+		else if (npcDialog != null)
+		{
+			String npcText = Text.sanitizeMultilineText(npcDialog.getText());
+			final Matcher mPermission = FOREMAN_PERMISSION_PATTERN.matcher(npcText);
+
+			if (mPermission.find())
+			{
+				foremanCounter = 100 * 10;
```

Comment:
Define the duration as a final static variable, and use a `Duration` instead of measuring in ticks.

Reply:
Resolved in 0fadcde